### PR TITLE
Implement split(ShapeContour, ShapeContour)

### PR DESCRIPTION
**Pull request by hamoid**

### CODE REVIEW

File: openrndr-core/src/main/kotlin/org/openrndr/shape/ShapeOperations.kt [hunk 1]
```diff
@@ -606,4 +606,38 @@ fun split(shape: Shape, line: LineSegment): Pair<Shape, Shape> {
...
     val rightShape = difference(shape, rightContour)
     return Pair(leftShape, rightShape)
+}
+
+/**
+ * Splits a ShapeContour with another ShapeContour.
+ * If there is no intersection it returns the original contour.
+ */
+fun split(from: ShapeContour, cutter: ShapeContour): List<ShapeContour> {
+    if (from.empty || cutter.empty)
```
Comment (edwinRNDR) [inline openrndr-core/src/main/kotlin/org/openrndr/shape/ShapeOperations.kt:616]: I think when the cutter is empty this code should return `listOf(from)`

Comment (hamoid) [reply openrndr-core/src/main/kotlin/org/openrndr/shape/ShapeOperations.kt]: Agree. Done :)

**Comment by hamoid** [comment]
Maybe you can wait before merge? I would like to add a method to split one contour with many contours. Would be more efficient to first find all intersections and then do the cuts at once, instead of calling the current .split many times.

**Comment by hamoid** [comment]
Second method added to be able to split against a list of ShapeContour instead of just one ShapeContour

**Comment by edwinRNDR** [comment]
So this is ready to be merged for you?

**Comment by hamoid** [comment]
Yes, on my tests it's working fine.

**Pull request merged**